src/djc_ext_pydantic/validation.py

```diff
@@ -0,0 +1,107 @@
+import sys
+from typing import Any, Literal, Optional, Tuple, Type, Union
+from weakref import WeakKeyDictionary
+
+from django_components import Component
+from pydantic import TypeAdapter, ValidationError
+
+
+ComponentTypes = Tuple[Any, Any, Any, Any, Any, Any]
+
+
+# Cache the types for each component class.
+# NOTE: `WeakKeyDictionary` can't be used as generic in Python 3.8
+if sys.version_info >= (3, 9):
+    types_store: WeakKeyDictionary[
+        Type[Component],
+        Union[Optional[ComponentTypes], Literal[False]],
+    ] = WeakKeyDictionary()
+else:
+    types_store = WeakKeyDictionary()
+
+
+def get_component_typing(cls: Type[Component]) -> Optional[ComponentTypes]:
```

Comment:
This file, together with `on_component_input` and `on_component_data` in `src/djc_ext_pydantic/extension.py` are practically what was on the Component class previously ([see here](https://github.com/django-components/django-components/blob/48dd3b7a5ac5437fa70c5757c408ff51e7ea43d7/src/django_components/component.py#L1554))